packages/template-retail-react-app/app/pages/login/index.jsx

```diff
@@ -29,10 +30,22 @@ const Login = () => {
     const einstein = useEinstein()
     const {isRegistered} = useCustomerType()
     const login = useAuthHelper(AuthHelpers.LoginRegisteredUserB2C)
+    const {mergeBasketIfNeeded} = useMergeBasket()
 
     const submitForm = async (data) => {
         try {
-            await login.mutateAsync({username: data.email, password: data.password})
+            await login.mutateAsync(
+                {username: data.email, password: data.password},
+                {
+                    onError: (error) => {
+                        const message = /Unauthorized/i.test(error.message)
+                            ? formatMessage(LOGIN_ERROR_MESSAGE)
+                            : formatMessage(API_ERROR_MESSAGE)
+                        form.setError('global', {type: 'manual', message})
+                    }
```

Comment:
We already have this same logic in the `catch` block.

Reply:
Oops, this is a bad merge. Good catch on that